sdks/python/apache_beam/io/fileio.py

```diff
@@ -796,17 +796,24 @@ def __iter__(self):
         # that the end offset of the range may be smaller than the original
         # end offset defined when creating the reader due to reader accepting
         # a dynamic split request from the service.
-        return
+        break
```

Comment:
This will yield one more line as a result of this change after `try_claim` fails, is this intentional?